aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -146,7 +147,7 @@ public static RestoreDbInstanceFromDbSnapshotRequest restoreDbInstanceFromSnapsh
                 .tdeCredentialArn(model.getTdeCredentialArn())
                 .tdeCredentialPassword(model.getTdeCredentialPassword())
                 .useDefaultProcessorFeatures(model.getUseDefaultProcessorFeatures())
-                .vpcSecurityGroupIds(model.getVPCSecurityGroups())
+                .vpcSecurityGroupIds(CollectionUtils.isNotEmpty(model.getVPCSecurityGroups()) ? model.getVPCSecurityGroups() : null)
```

Comment:
Unfornately, putting `vpcSecurityGroupIds` with null will be converted to empty list as well.

here is the implementation
```
        public final ModifyDbInstanceRequest.Builder vpcSecurityGroupIds(Collection<String> vpcSecurityGroupIds) {
            this.vpcSecurityGroupIds = VpcSecurityGroupIdListCopier.copy(vpcSecurityGroupIds);
            return this;
        }
```
and
```
final class VpcSecurityGroupIdListCopier {
    VpcSecurityGroupIdListCopier() {
    }

    static List<String> copy(Collection<String> vpcSecurityGroupIdListParam) {
        Object list;
        if (vpcSecurityGroupIdListParam != null && !(vpcSecurityGroupIdListParam instanceof SdkAutoConstructList)) {
            List<String> modifiableList = new ArrayList();
            vpcSecurityGroupIdListParam.forEach((entry) -> {
                modifiableList.add(entry);
            });
            list = Collections.unmodifiableList(modifiableList);
        } else {
            list = DefaultSdkAutoConstructList.getInstance();
        }

        return (List)list;
    }
}
```

Reply:
I know, this is the idea. The `SdkAutoConstructList` is getting serialized into null rather than an empty list (like `ArrayList`)